### ISSUE TITLE
Microsoft TTS

### ISSUE THREAD

**Issue by PointPubMedia** [jishi/node-sonos-http-api#271]
@jishi Do you think it's possible to have Microsoft TTS? ;)

**Comment by PointPubMedia** [comment]
@jishi Willing to pay for Microsoft TTS + Multiple zone in say command ;)

**Comment by jishi** [comment]
Depends on how it works, do you have a link or documentation for it?

**Comment by PointPubMedia** [comment]
@jishi I can try to find some for you!

**Comment by PointPubMedia** [comment]
@jishi https://www.microsoft.com/en-us/translator/getstarted.aspx

**Comment by jishi** [comment]
Have you tested their TTS service? What is the benefit of it? Seems kind of
complex to setup since it require app registration etc...

**Comment by PointPubMedia** [comment]
@jishi I'm using them since a year with another plugin for home automation.  The TTS output is way better than the one from voicerss!

So your awesome code with MS TTS will be kicking ass ;)

**Comment by PointPubMedia** [comment]
@jishi The voice is more fluid and more natural!

**Comment by PointPubMedia** [comment]
@jishi Can we have a chat by email ?

**Comment by jishi** [comment]
You can reach me at jimmy at shimizu.se

**Comment by PointPubMedia** [comment]
@jishi , any update about this feature? ;)

**Comment by jishi** [comment]
No, I'm sorry to say that this have very low priority. I'm considering adding back Googles TTS as a free alternative, although it will cap you if you request to many audio clips.

**Comment by PointPubMedia** [comment]
MS TTS it's free too and have a very high limit per free project...

Personnaly, I way prefer MS TTS output as Google!   Was using MS TTS before switch to your node-sonos-http-api for our home automation playback!

**Comment by jishi** [comment]
I did some tests with the Bing Text To Speech API (I assume) and it seems fairly trivial to get this working. I will give it a try in parallel with the AWS Polly service to give people more and better options.

I'll keep this issue updated when there is progress.

**Comment by jishi** [comment]
Okay, the latest master branch now has microsoft support. Please evaluate it and leave feedback. I wrote down instructions in the README, let me know if they are easy to understand or need improvements.

**Comment by PointPubMedia** [comment]
Just made the switch, test a bit and it's awesome ;)
Thanks a lot!

**Comment by PointPubMedia** [comment]
Found a bug ;)
When using the Microsoft TTS and the SAYALL, the TTS convert space and other special caracter into %something and say them...

**Comment by jishi** [comment]
Forgot to adjust the sayall and didn't unescape the text phrase. I committed a fix directly from github, test it out and see if it works.

**Comment by PointPubMedia** [comment]
Jimmy,

Looks like it's not working at 100% for the sayall, less issue but not 
as good as without sayall!


Sébastien Roy
Conseiller Médias / Spécialiste Infrastructure

PointPub Media Communications. - St-Eustache, Canada
Agence Web -- Services Internet -- Services Affaires
Bur.: (514) 312-7227, x5201

**Comment by PointPubMedia** [comment]
I just test it and looks like the %20 are ok now... but it having issue 
with special caracter... like é

Sébastien Roy
Conseiller Médias / Spécialiste Infrastructure

PointPub Media Communications. - St-Eustache, Canada
Agence Web -- Services Internet -- Services Affaires
Bur.: (514) 312-7227, x5201

**Comment by jishi** [comment]
I'll have a look later tonight. The documentation from Microsoft was a bit unclear on how they expected it to be encoded. It said URL encoded but apparently it choke on %20.
